src/github-actions/actions.ts

```diff
@@ -0,0 +1,56 @@
+import { ActionMetadata, RunsUsing } from "../github/actions-metadata-model";
+import { TypeScriptProject, TypeScriptProjectOptions } from "../typescript";
+import { YamlFile } from "../yaml";
+import { GitHubActionMetadata } from "./github-action-metadata";
+
+/**
+ * Properties for creating a GitHubActionTypeScriptProject
+ */
+export interface GitHubActionTypeScriptOptions
+  extends TypeScriptProjectOptions {
+  /**
+   * Every GitHub Action must have a metadata file named `action.yml`.
+   * Projen will manage this file for you using the specifications of
+   * this property.
+   */
+  readonly metadata?: GitHubActionMetadata;
```

Comment:
Suggestion to protect API surface area from future name conflicts / be more explicit
```suggestion
  readonly actionMetadata?: GitHubActionMetadata;
```